src/Turing.jl

```diff
@@ -37,22 +37,27 @@ include("stdlib/distributions.jl")
 include("stdlib/RandomMeasures.jl")
 
 """
-struct Model{F, Targs <: NamedTuple}
+struct Model{F, Targs <: NamedTuple, Tmissings <: Val}
     f::F
     args::Targs
+    missings::Tmissings
 end
 
 A `Model` struct with arguments `args` and inner function `f`.
 """
-struct Model{F, Targs <: NamedTuple} <: AbstractModel
+struct Model{F, Targs <: NamedTuple, Tmissings <: Val} <: AbstractModel
     f::F
     args::Targs
+    missings::Tmissings
 end
+Model(f, args::NamedTuple) = Model(f, args, getmissing(args))
 (model::Model)(vi) = model(vi, SampleFromPrior())
 (model::Model)(vi, spl) = model(vi, spl, DefaultContext())
 (model::Model)(args...; kwargs...) = model.f(args..., model; kwargs...)
-getmissing(model::Model) = _getmissing(model.args)
-@generated function _getmissing(args::NamedTuple{names, ttuple}) where {names, ttuple}
+
+getmissing(model::Model) = model.missings
+@generated function getmissing(args::NamedTuple{names, ttuple}) where {names, ttuple}
+    length(names) == 0 && return :(Val{()}())
     minds = filter(1:length(names)) do i
```

Comment:
I'm confused by this syntax. Sorry for the stupid question, but what does `filter(1:length(names))` do? Is this not equivalent to `1:length(names)`?

Reply:
The `do` syntax is a convenient way to use longer function definitions, this is equivalent to `filter(i -> ...., 1:length(names))`.

Reply:
In this case it is not really long, but I find it more readable.